kolena/_experimental/quality_standard.py

```diff
@@ -72,12 +97,195 @@ def _download_quality_standard_result(
+            ),
+        ),
+        api_version="v2",
+    )
+    krequests.raise_for_status(response)
+    return from_dict(TestingResponse, response.json())
+
+
+def _download_quality_standard(
+    dataset_id: int,
+) -> QualityStandardResponse:
+    response = krequests.get(
+        Path.QUALITY_STANDARD,
+        params=dict(dataset_id=dataset_id),
+        api_version="v2",
+    )
+    krequests.raise_for_status(response)
+    result = response.json()
+    return from_dict(QualityStandardResponse, result)
+
+
+def _calculate_moe_map(
+    qs_result: pd.DataFrame,
+    dataset_entity: EntityData,
+    confidence_level: float,
+    qs: QualityStandardResponse,
+) -> Dict[Tuple[str, Any], float]:
+    model_name_to_id_map = {}
+    moe = {}
+    for model, eval_config, _, _ in qs_result.columns:
+        deserialized_eval_config = eval_config
+        if eval_config == "null":
+            deserialized_eval_config = None
+        if eval_config and eval_config != "null":
+            deserialized_eval_config = json.loads(eval_config)
+        model_id = _get_model_id(model)
+        eval_config_id = _get_eval_config_id(deserialized_eval_config)
+        model_name_to_id_map[(model, eval_config)] = ModelWithEvalConfig(model_id, eval_config_id)
+
+    for strat in qs.quality_standard.stratifications:
+        test_cases = _download_test_case_result(
+            dataset_entity.id,
+            list(model_name_to_id_map.values()),
+            strat.filters,
+            strat.stratify_fields,
+        )
+        for i in range(len(test_cases.test_cases)):
+            case = test_cases.test_cases[i]
+            if not case.stratification or not case.stratification[0]:
+                continue
+            value = case.stratification[0].value
+            field = case.stratification[0].field
+            if field.startswith("_kolena.extracted"):
+                field_keys = field.split(".")
+                field = ".".join(field_keys[3:])
+            if case.stratification[0].type in {StratificationType.EQUAL_HEIGHT, StratificationType.EQUAL_WIDTH}:
+                value = f"{ordinal(i + 1)} {get_quantile(len(test_cases.test_cases))}"
+            moe[(field, value)] = margin_of_error(case.sample_count, confidence_level)
+        if ("Dataset", np.nan) not in moe:
+            moe[("Dataset", np.nan)] = margin_of_error(test_cases.dataset.sample_count, confidence_level)
+    return moe
+
+
+def _get_performance_delta_metrics(
+    qs: QualityStandardResponse,
+    qs_result: pd.DataFrame,
+    moe: Dict[Tuple[str, Any], float],
+    reference_model: Optional[str] = None,
+    reference_eval_config: Union[Dict[str, Any], Literal["null"], None] = None,
+) -> pd.DataFrame:
+    for model, eval_config, _, _ in qs_result.columns:
+        if not reference_model:
+            reference_model = model
+        if reference_model == model and not reference_eval_config:
+            reference_eval_config = eval_config
```

Comment:
Does this mean that the first `eval_config` will be skipped if it is `None`?

```
# initially reference_model is None and reference_eval_config
# First pass, model is 'foo' and eval_config is None
reference_model = 'foo'
reference_eval_config = None
# Second pass, model is 'foo' and eval_config is {'bar': 'baz'}
reference_eval_config = {'bar': 'baz'}
```

Also, does this get handled safely if the only eval config for the reference model is `None`?

Reply:
originally I was expecting user to pass in "null" so it won't be ignored, I will have to handle this case now. This should work as expected when the only eval config is None, as the dataset I used to test is like this